modules/dmrpp_module/build_dmrpp_util.cc

```diff
@@ -1161,50 +1165,243 @@ void add_chunk_information(const string &h5_file_name, DMRpp *dmrpp)
     }
 }
 
+
 /**
- * @brief Write information to the the DMR++ about its provenance.
- * @param dmrpp Add provenance information to this instance of DMRpp
- * @note The DMRpp instance will free all memory allocated by this method.
+ * @brief Performs a quality control check on the user supplied data file.
+ *
+ * The supplied file is going to be used by build_dmrpp as the source of variable/dataset chunk information.
+ * At the time of this writing only netcdf-4 and hdf5 file encodings are supported (Note that netcdf-4 is a subset of
+ * hdf5 and all netcdf-4 files are defacto hdf5 files.)
+ *
+ * To that end this function will:
+ * * Test that the file exists and can be read from.
+ * * The first few bytes of the file will be checked to ensure that it is an hdf5 file.
+ * * If it's not an hdf5 file the head bytes will checked to see if the file is a netcdf-3 file, as that is common
+ *   mistake.
+ *
+ * @param file_name
  */
-    void inject_version_and_configuration(DMRpp *dmrpp)
-    {
-        bool found;
+void qc_input_file(const string &file_fqn)
+{
+    //Use an ifstream file to run a check on the provided file's signature
+    // to see if it is an HDF5 file. - kln 5/18/23
+
+    if (file_fqn.empty()) {
+        stringstream msg;
+        msg << "HDF5 input file name must be provided (-f <input>) and be a fully qualified path name." << endl;
+        throw BESInternalFatalError(msg.str(), __FILE__, __LINE__);
+    }
 
-        dmrpp->set_version(CVER);
+    std::ifstream file(file_fqn, ios::binary);
+    auto errnum = errno;
+    if (!file)  // This is same as if(file.fail()){...}
+    {
+        stringstream msg;
+        msg << "Encountered a Read/writing error when attempting to open the file: " << file_fqn << endl;
+        msg << "*  strerror(errno): " << strerror(errnum) << endl;
+        msg << "*          failbit: " << (((file.rdstate() & std::ifstream::failbit) != 0) ? "true" : "false") << endl;
+        msg << "*           badbit: " << (((file.rdstate() & std::ifstream::badbit) != 0) ? "true" : "false") << endl;
+        msg << "Things to check:" << endl;
+        msg << "* Does the file exist at expected location?" << endl;
+        msg << "* Does your user have permission to read the file?" << endl;
+        throw BESInternalFatalError(msg.str(), __FILE__, __LINE__);
+    }
 
-        // Build the version attributes for the DMR++
-        auto version = new D4Attribute("build_dmrpp_metadata", StringToD4AttributeType("container"));
+    //HDF5 and NetCDF3 signatures:
+    const char hdf5Signature[] = {'\211', 'H', 'D', 'F', '\r', '\n', '\032', '\n'};
+    const char netcdf3Signature[] = {'C', 'D', 'F'};
 
-        auto build_dmrpp_version = new D4Attribute("build_dmrpp", StringToD4AttributeType("string"));
-        build_dmrpp_version->add_value(CVER);
-        version->attributes()->add_attribute_nocopy(build_dmrpp_version);
+    //Read the first 8 bytes (file signature) from the file
+    char signature[8];
```

Comment:
You can use std::string here instead of a c-style array.  I'm pretty sure you can say:
string signature;
signature.reserve(8);

file.read(signature.data(), 8)

Is this better? Well, code scanners will gripe about C-style arrays, and knowing how to mix the C++ strings with code that was designed with C-arrays in mind is probably a good thing. I don't know if it's really an improvement... Your call on this change.

You can also do this using:
string signature(8);

But, it's important to use ()'s there and not braces.